compiler/rustc_parse_format/src/lib.rs

```diff
@@ -224,7 +224,7 @@ impl<'a> Iterator for Parser<'a> {
                 '{' => {
                     let curr_last_brace = self.last_opening_brace;
                     let byte_pos = self.to_span_index(pos);
-                    let lbrace_end = InnerOffset(byte_pos.0 + 1);
+                    let lbrace_end = self.to_span_index(pos + 1);
```

Comment:
There is a manual creation of an `InnerOffset` just below (github won't let me comment on it).  Could it trigger a similar bug?  Should it be changed in a similar way?

Reply:
That's a good point. Originally I had assumed not since I wasn't able to find any breaking examples, but I just found that following:

```rust
println!("{\x7D");
```

gives the following error:

```
error: 1 positional argument in format string, but no arguments were given
 --> src/main.rs:2:15
  |
2 |     println!("{\x7D");
  |  
```

I can go ahead and apply the fix to that line as well, and add some more tests.

Reply:
Yes, please, with the test if possible.  That's a good opportunity to fix several identical bugs at once.  I'll approve once this is done.

Reply:
Fixed in efc6df2.